pkg/dmlegacy/image_format.go

```diff
@@ -82,7 +82,10 @@ func addFiles(img *api.Image, src source.Source) error {
 	if _, err := util.ExecuteCommand("mount", "-o", "loop", p, tempDir); err != nil {
 		return fmt.Errorf("failed to mount image %q: %v", p, err)
 	}
-	defer util.ExecuteCommand("umount", tempDir)
+	defer util.DeferErr(&err, func() error {
+		_, err := util.ExecuteCommand("umount", tempDir)
```

Comment:
this seems a bit dangerous, to use the same `err` variable name

Reply:
Right, I'll change it. Thanks.